backend/src/dao/TeamEventAttendanceDao.ts

```diff
@@ -1 +1,44 @@
-export default class TeamEventAttendanceDao {}
+import { v4 as uuidv4 } from 'uuid';
+import { memberCollection, teamEventAttendanceCollection } from '../firebase';
+import { DBTeamEventAttendance } from '../types/DataTypes';
+
+export default class TeamEventAttendanceDao {
+  static async createTeamEventAttendance(
+    teamEventAttendance: TeamEventAttendance
+  ): Promise<TeamEventAttendance> {
+    const teamAttendanceRef: DBTeamEventAttendance = {
+      member: memberCollection.doc(teamEventAttendance.member.email),
+      ...(typeof teamEventAttendance.hoursAttended === 'number'
+        ? { hoursAttended: teamEventAttendance.hoursAttended }
+        : {}),
+      image: teamEventAttendance.image,
+      eventUuid: teamEventAttendance.eventUuid,
+      pending: true,
+      uuid: teamEventAttendance.uuid ? teamEventAttendance.uuid : uuidv4()
+    };
+    await teamEventAttendanceCollection.doc(teamAttendanceRef.uuid).set(teamAttendanceRef);
+    return teamEventAttendance;
+  }
+
+  static async updateTeamEventAttendance(
+    attendance: TeamEventAttendance
```

Comment:
This is my fault for writing the specs this way. I called the parameter here `attendance` but for `createTeamEventAttendance` I called it `teamEventAttendance`. We should have consistency everywhere this is used (and we haven't done a great job in some of our other dao classes, TBH). Let's call it `teamEventAttendance` everywhere. I just updated this in the ticket and the Design Doc. Again, this one's my fault.